### ISSUE TITLE
Move temporary files in a TMPDIR subfolder

### ISSUE THREAD

**Issue by 3v1n0** [jisaacks/GitGutter#483]
## Description

Gitgutter generates lots of temporary files in $TMPDIR, it would be nicer if they could be moved inside a directory, not to clutter the contents of that dir with too much data.

```bash
$ ls /tmp/git_gutter_*
/tmp/git_gutter_07a3g1  /tmp/git_gutter_97tpq0  /tmp/git_gutter_d2mlpd  /tmp/git_gutter_k9pj7y  /tmp/git_gutter_pjlu5t  /tmp/git_gutter_w9auv6
/tmp/git_gutter_0erf6f  /tmp/git_gutter_a7p05u  /tmp/git_gutter_e9enrz  /tmp/git_gutter_l6ar5i  /tmp/git_gutter_q488p7  /tmp/git_gutter_xwnxeu
/tmp/git_gutter_23isv2  /tmp/git_gutter_ammoi6  /tmp/git_gutter_esaa_s  /tmp/git_gutter_l9qdcy  /tmp/git_gutter_q6gx_t  /tmp/git_gutter_y5n9j2
/tmp/git_gutter_51ophj  /tmp/git_gutter_aoxt4d  /tmp/git_gutter_gbzm33  /tmp/git_gutter_likxow  /tmp/git_gutter_tmmv_i  /tmp/git_gutter_yky48u
/tmp/git_gutter_59goca  /tmp/git_gutter_b5aedl  /tmp/git_gutter_hb_ons  /tmp/git_gutter_mfoax_  /tmp/git_gutter_twxbsm
/tmp/git_gutter_61kjmq  /tmp/git_gutter_bcfvsj  /tmp/git_gutter_iw0pl0  /tmp/git_gutter_mosxwl  /tmp/git_gutter_txoon6
/tmp/git_gutter_6xollp  /tmp/git_gutter_cvp0mp  /tmp/git_gutter_j6_y9e  /tmp/git_gutter_oon5ye  /tmp/git_gutter_vc3idy
/tmp/git_gutter_8ewlu5  /tmp/git_gutter_d1taxs  /tmp/git_gutter_jmuj88  /tmp/git_gutter_p7jx45  /tmp/git_gutter_vl7au3
```

**Comment by andyneff** [comment]
That would be a nice idea. Why aren't any of these cleaned up? Wasn't this fixed in the past already? #284 #247 

Same problem here running Fedora 27 and Sublime 3 3143

**Comment by deathaxe** [comment]
They normally should, but python sometimes doesn't seem to clean up objects and therefore files are not deleted sometimes. Under normal conditions files are cleaned up correctly (at least on Windows) but from time to time I found some zombies left behind. Mostly after I changed some pieces of code and reloaded GitGutter.

**Comment by deathaxe** [comment]
This issue rises some more questions, which may be worth discussing:

1. On Windows the temporary directory is located within the user profile path and therefore private for each user. On Unix/Linux it looks like the global `/tmp` folder is used. So the temporary files are public for each user being logged on.

2. Even with "portable" setup, GitGutter causes files being placed outside the application directory, which somehow breaks the idea of "portable". 

Therefore I am wondering whether it was a better idea to place temporary files into ST's _Data/Cache/GitGutter_ folder, which would solve both issues.

**Comment by deathaxe** [comment]
Fixed in 1.7.12
